### PR TITLE
added process to auxiliary process list

### PR DESCRIPTION
**Pull request by aronnoordam**

### CODE REVIEW

File: benchmark_tests/test_mass_on_spring_damper/test_mass_on_spring_damper.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+from stem.boundary import DisplacementConstraint
+from stem.solver import (AnalysisType, SolutionType, TimeIntegration, DisplacementConvergenceCriteria,
+                         StressInitialisationType, SolverSettings, Problem)
+from stem.output import NodalOutput, GaussPointOutput, VtkOutputParameters, Output
+from stem.stem import Stem
+from benchmark_tests.utils import assert_files_equal
+from shutil import rmtree
+
+
+def test_stem():
```
Comment (brunozc) [inline benchmark_tests/test_mass_on_spring_damper/test_mass_on_spring_damper.py:16]: work in progress

File: stem/IO/kratos_io.py [hunk 2]
```diff
@@ -962,11 +1016,13 @@ def write_project_parameters_json(self, model: Model, mesh_file_name: str,
         outputs_dict = self.__create_output_process_dictionary(outputs=model.output_settings)
         # get the boundary condition dictionary
         loads_and_bc_dict = self.__create_loads_and_boundary_conditions_dictionary(model=model)
+        # get the auxiliary processes dictionary
```
Comment (brunozc) [inline stem/IO/kratos_io.py:1019]: why are the mass spring system called auxiliar?

File: stem/IO/kratos_material_io.py [hunk 3]
```diff
@@ -380,8 +379,7 @@ def create_material_dict(
         # initialize material dictionary
         material_dict: Dict[str, Any] = {"model_part_name": f"{self.domain}.{part_name}",
                                          "properties_id": material_id,
-                                         "Material": {"constitutive_law": {"name": ""},
```
Comment (brunozc) [inline stem/IO/kratos_material_io.py:383]: is this no longer needed?

File: tests/test_kratos_io.py [hunk 4]
```diff
@@ -486,4 +508,99 @@ def test_write_mdpa_two_conditions_same_position(self):
...
+        model = create_default_2d_model
+        kratos_io = KratosIO(ndim=model.ndim)
+        model.project_parameters = create_default_solver_settings
+        model.output_settings = create_default_outputs
+
+        # add elastic spring damper element
+        spring_damper = ElasticSpringDamper(
+            NODAL_DISPLACEMENT_STIFFNESS=[1, 1, 1],
+            NODAL_ROTATIONAL_STIFFNESS=[1, 1, 2],
+            NODAL_DAMPING_COEFFICIENT=[1, 1, 3],
```
Comment (brunozc) [inline tests/test_kratos_io.py:535]: are you sure it is damping coefficient? units [N s/m]

File: stem/IO/kratos_io.py (hunk 2 above)
Comment (aronnoordam) [reply stem/IO/kratos_io.py:1019]: the mass and spring systems are still elements (body model parts), however a different process was needed to assign the parameters to the elements. This process is implemented in StemKratos

File: stem/IO/kratos_material_io.py (hunk 3 above)
Comment (aronnoordam) [reply stem/IO/kratos_material_io.py:383]: not as default, spring damper and mass elements dont have a separate constitutive law in Kratos. (the law is incorporated in the element)

File: tests/test_kratos_io.py (hunk 4 above)
Comment (aronnoordam) [reply tests/test_kratos_io.py:535]: Yes, it is wrongly name in Kratos, the damping matrix in kratos is build as follows:
`C = [[c, -c],[-c, c]`

**Review by brunozc — APPROVED** [review]

**Pull request merged**